packages/ai-chat-ui/src/browser/chat-tree-view/chat-view-tree-widget.tsx

```diff
@@ -389,6 +377,27 @@ export class ChatViewTreeWidget extends TreeWidget {
     }
 }
 
+const ChatRequestRender = ({ node }: { node: RequestNode }) => {
+    // eslint-disable-next-line no-null/no-null
+    const ref: React.MutableRefObject<HTMLDivElement | null> = useRef(null);
+    useEffect(() => {
+        const markdownIt = markdownit();
+        const text = node.request.request.displayText ?? node.request.request.text;
+        const host = document.createElement('div');
+        const html = markdownIt.render(text);
+        host.innerHTML = DOMPurify.sanitize(html, {
+            ALLOW_UNKNOWN_PROTOCOLS: true // DOMPurify usually strips non http(s) links from hrefs
+        });
+        while (ref?.current?.firstChild) {
+            ref.current.removeChild(ref.current.firstChild);
+        }
+
+        ref?.current?.appendChild(host);
+    }, [node.request]);
```

Comment:
This is almost the same content as in the response renderer. I would like to suggest a custom hook `useMarkDownRendering(markDownText)` so we can reuse this logic in both.

The hook could then also be documented on why it exists and why it uses markdownit directly.